cosmwasm/submit-proposal.js

```diff
@@ -195,53 +190,58 @@ const main = async (options) => {
-                        return promise.then(() => storeInstantiate(client, wallet, config, options, chain.toLowerCase()));
-                    }, Promise.resolve());
-                }
+        case 'storeInstantiate': {
+            const chains = getChains(config, options);
 
-                case 'instantiate': {
-                    const chains = getChains(config, options);
-
-                    return chains.reduce((promise, chain) => {
-                        return promise.then(() =>
-                            instantiate(client, wallet, config, options, chain.toLowerCase()).then((contractAddress) => {
-                                if (contractAddress) {
-                                    printInfo(
-                                        `Predicted address for ${
-                                            chain.toLowerCase() === 'none' ? '' : chain.toLowerCase().concat(' ')
-                                        }${contractName}. Address`,
-                                        contractAddress,
-                                    );
-                                }
-                            }),
-                        );
-                    }, Promise.resolve());
-                }
+            for (const chain of chains) {
+                await storeInstantiate(client, wallet, config, options, chain.toLowerCase());
+            }
 
-                case 'execute': {
-                    const chains = getChains(config, options);
+            break;
+        }
 
-                    return chains.reduce((promise, chain) => {
-                        return promise.then(() => execute(client, wallet, config, options, chain.toLowerCase()));
-                    }, Promise.resolve());
+        case 'instantiate': {
+            const chains = getChains(config, options);
+
+            for (const chain of chains) {
+                const contractAddress = await instantiate(client, wallet, config, options, chain.toLowerCase());
+
+                if (contractAddress) {
+                    printInfo(
+                        `Predicted address for ${
+                            chain.toLowerCase() === 'none' ? '' : chain.toLowerCase().concat(' ')
+                        }${contractName}. Address`,
+                        contractAddress,
+                    );
                 }
+            }
+
+            break;
+        }
+
+        case 'execute': {
+            const chains = getChains(config, options);
 
-                default:
-                    throw new Error('Invalid proposal type');
+            for (const chain of chains) {
+                await execute(client, wallet, config, options, chain.toLowerCase());
             }
-        })
-        .then(() => saveConfig(config, env));
+
+            break;
+        }
+
+        default:
+            throw new Error('Invalid proposal type');
+    }
+
+    saveConfig(config, env);
 };
 
 const programHandler = () => {
```

Comment:
not in this PR, but you can refactor the CLI handler to use commands. So options can be defined only for the cmds that need them, similar to axelard CLI
https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/sui/gateway.js#L343-L358

Reply:
yeah, I have been wanting to do that, but haven't had the chance to dig deeper into how sub commands work, will take a look at the sui script